src/QueryPipeline/RemoteQueryExecutorReadContext.cpp

```diff
@@ -56,16 +57,32 @@ void RemoteQueryExecutorReadContext::Task::run(AsyncCallback async_callback, Sus
 
     while (true)
     {
-        read_context.has_read_packet_part = PacketPart::None;
-
-        if (read_context.read_packet_type_separately)
+        try
+        {
+            read_context.has_read_packet_part = PacketPart::None;
+
+            if (read_context.read_packet_type_separately)
+            {
+                read_context.packet.type = read_context.executor.getConnections().receivePacketTypeUnlocked(async_callback);
+                read_context.has_read_packet_part = PacketPart::Type;
+                suspend_callback();
+            }
+            read_context.packet = read_context.executor.getConnections().receivePacketUnlocked(async_callback);
+            read_context.has_read_packet_part = PacketPart::Body;
+            if (read_context.packet.type == Protocol::Server::Data)
+                read_context.has_data_packets = true;
+        }
+        catch (const Exception & e)
         {
-            read_context.packet.type = read_context.executor.getConnections().receivePacketTypeUnlocked(async_callback);
-            read_context.has_read_packet_part = PacketPart::Type;
-            suspend_callback();
+            if (e.code() == ErrorCodes::ATTEMPT_TO_READ_AFTER_EOF
```

Comment:
Could you please add a comment explaining what `ErrorCodes::ATTEMPT_TO_READ_AFTER_EOF` usually means and why you are catching it?

The same for the below if statement

Reply:
Added comment there.
It's required for non-autodiscovery case, when shutdowned node described in xml-config, in this case node still present in cluster after `stop swarm mode`.
In autodiscovery case node must be removed from cluster during `stop swarm mode` command.